src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(health = 100, name) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, hidden, health = 100) {
+    super(hidden);
+    this.hidden = false;
+    this.health = health;
+    this.name = name;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, type = 'carnivore') {
+    super(health, name);
+    this.health = health;
+    this.type = type;
+  }
+
+  bite(animal) {
+    if (animal.hidden !== true && animal.type !== 'carnivore') {
+      animal.health -= 50;
+    }
+
+    if (animal.health <= 0) {
+      animal.die();
+    }
```

Comment:
The 'die' method should not be called directly. Instead, filter the 'Animal.alive' array to remove the dead animal after its health drops below or equal to 0.